### ISSUE TITLE
Improve error message when UUID could not be found

### ISSUE THREAD

**Issue by yangfl** [nextcloud/server#27859]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Setup an LDAP server with some user objects, but avoid any attributes in https://github.com/nextcloud/server/blob/db0bda06b55c997a776b8580cbefc54d5d76b266/apps/user_ldap/lib/Access.php#L69
2. Set up LDAP connection as usual, but do not touch any expert settings.

### Expected behaviour
You should be told 'Override UUID detection' (`ldapExpertUUIDUserAttr`) is missing.

### Actual behaviour
The error log is 

> LDAP Login: Could not get user object for DN uid=xx,ou=xx,dc=xx. Maybe the LDAP entry has no set display name attribute?

However, this is nothing about 'User Display Name Field' in 'Advanced settings'.

Also, the term UUID is misleading since it's not the well-known 'Universally unique identifier', which is a 128-bit label and usually of form 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'. Actually, it accepts free-form strings as long as it is unique among users. Please avoid confusing abbreviation.

### Server configuration

**Operating system:** Debian 11

**Web server:** nginx

**Database:** mariadb

**PHP version:** 7.4

**Nextcloud version:** 21.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** tarball

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - bruteforcesettings: 2.2.0
  - calendar: 2.3.0
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - mail: 1.10.0
  - nextcloud_announcements: 1.10.0
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - recommendations: 1.0.0
  - richdocuments: 4.2.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - spreed: 11.2.2
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - text: 3.2.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_ldap: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - admin_audit
  - encryption
  - files_external
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "127.0.0.1"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "21.0.3.1",
        "overwrite.cli.url": "http:\/\/127.0.0.1:8080\/nextcloud",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "allow_local_remote_servers": true,
        "maintenance": false
    }
}
```
</details>

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
+-------------------------------+------------------------------------------------------+
| Configuration                 | s01                                                  |
+-------------------------------+------------------------------------------------------+
| hasMemberOfFilterSupport      | 0                                                    |
| homeFolderNamingRule          |                                                      |
| lastJpegPhotoLookup           | 0                                                    |
| ldapAgentName                 | cn=xx,ou=xx,dc=xx                 |
| ldapAgentPassword             | ***                                                  |
| ldapAttributesForGroupSearch  |                                                      |
| ldapAttributesForUserSearch   |                                                      |
| ldapBackupHost                |                                                      |
| ldapBackupPort                |                                                      |
| ldapBase                      | dc=xx                                  |
| ldapBaseGroups                | ou=xx,dc=xx                         |
| ldapBaseUsers                 | ou=xx,dc=xx                        |
| ldapCacheTTL                  | 600                                                  |
| ldapConfigurationActive       | 1                                                    |
| ldapDefaultPPolicyDN          |                                                      |
| ldapDynamicGroupMemberURL     |                                                      |
| ldapEmailAttribute            | mail                                                 |
| ldapExperiencedAdmin          | 1                                                    |
| ldapExpertUUIDGroupAttr       | cn                                                   |
| ldapExpertUUIDUserAttr        | uid                                                  |
| ldapExpertUsernameAttr        |                                                      |
| ldapExtStorageHomeAttribute   |                                                      |
| ldapGidNumber                 | gidNumber                                            |
| ldapGroupDisplayName          | cn                                                   |
| ldapGroupFilter               |                                                      |
| ldapGroupFilterGroups         |                                                      |
| ldapGroupFilterMode           | 0                                                    |
| ldapGroupFilterObjectclass    |                                                      |
| ldapGroupMemberAssocAttr      |                                                      |
| ldapHost                      | localhost                                            |
| ldapIgnoreNamingRules         |                                                      |
| ldapLoginFilter               | (&(objectclass=nextcloudUser)(uid=%uid))             |
| ldapLoginFilterAttributes     |                                                      |
| ldapLoginFilterEmail          | 0                                                    |
| ldapLoginFilterMode           | 0                                                    |
| ldapLoginFilterUsername       | 1                                                    |
| ldapMatchingRuleInChainState  | unknown                                              |
| ldapNestedGroups              | 0                                                    |
| ldapOverrideMainServer        |                                                      |
| ldapPagingSize                | 500                                                  |
| ldapPort                      | 389                                                  |
| ldapQuotaAttribute            | nextcloudQuota                                       |
| ldapQuotaDefault              |                                                      |
| ldapTLS                       | 0                                                    |
| ldapUserAvatarRule            | default                                              |
| ldapUserDisplayName           | cn                                                   |
| ldapUserDisplayName2          |                                                      |
| ldapUserFilter                | (&(objectclass=nextcloudUser)(nextcloudEnabled=yes)) |
| ldapUserFilterGroups          |                                                      |
| ldapUserFilterMode            | 0                                                    |
| ldapUserFilterObjectclass     | nextcloudUser                                        |
| ldapUuidGroupAttribute        | auto                                                 |
| ldapUuidUserAttribute         | auto                                                 |
| turnOffCertCheck              | 0                                                    |
| turnOnPasswordChange          | 0                                                    |
| useMemberOfToDetectMembership | 1                                                    |
+-------------------------------+------------------------------------------------------+
```
</details>

**Comment by szaimen** [comment]
cc @nextcloud/ldap is this feasible?

**Comment by wittj** [comment]
I ran into the same issue when integrating Nextcloud 25.0.2 via LDAP provided by IDP Authentik (https://goauthentik.io). The error log showing "Maybe the LDAP entry has no set display name attribute?" as a hint to fix the error was misleading to me. Thanks to this issue here I found the correct solution to solve the problem.

So a reassessment of this issue if the log message could be improved could be very helpful for other users running into the same problem.
